os/arch/arm/include/amebalite/irq.h

```diff
@@ -0,0 +1,160 @@
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#define NR_IRQS 100
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+
+/* IRQ numbers.  The IRQ number corresponds vector number and hence map directly to
+ * bits in the NVIC.  This does, however, waste several words of memory in the IRQ
+ * to handle mapping tables.
+ */
+
+/* Processor Exceptions (vectors 0-15) */
+
+#define AMEBALITE_IRQ_RESERVED       (0) /* Reserved vector (only used with CONFIG_DEBUG) */
+                                     /* Vector  0: Reset stack pointer value */
+                                     /* Vector  1: Reset (not handler as an IRQ) */
+#define AMEBALITE_IRQ_NMI            (2) /* Vector  2: Non-Maskable Interrupt (NMI) */
+#define AMEBALITE_IRQ_HARDFAULT      (3) /* Vector  3: Hard fault */
+#define AMEBALITE_IRQ_MEMFAULT       (4) /* Vector  4: Memory management (MPU) */
+#define AMEBALITE_IRQ_BUSFAULT       (5) /* Vector  5: Bus fault */
+#define AMEBALITE_IRQ_USAGEFAULT     (6) /* Vector  6: Usage fault */
+                                     /* Vectors 7-10: Reserved */
+#define AMEBALITE_IRQ_SVCALL        (11) /* Vector 11: SVC call */
+#define AMEBALITE_IRQ_DBGMONITOR    (12) /* Vector 12: Debug Monitor */
+                                     /* Vector 13: Reserved */
+#define AMEBALITE_IRQ_PENDSV        (14) /* Vector 14: Pendable system service request */
+#define AMEBALITE_IRQ_SYSTICK       (15) /* Vector 15: System tick */
+
+/* External interrupts (vectors >= 16).  These definitions are chip-specific */
+
+#define AMEBALITE_IRQ_FIRST         (16) /* Vector number of the first external interrupt */
+
+#define AMEBALITE_IRQ_WIFI_FISR_FESR				(AMEBALITE_IRQ_FIRST + 0)

```

Comment:
Could  you modify the tab spaces so that all Macro definitions have the same indentation please?

Reply:
@amogh-samsung We don't guide to follow TizenRT coding rule for chipset codes like arch, board and so on. How about keeping them as the chipset vendor does?

Reply:
yes we agreed to give exception for vendor code, let vendor follow their standard.